### DISCUSSION TITLE
Content Security Policy Cached on Browsers

### DISCUSSION THREAD

**Discussion by ashbeech** [vercel/community#4015] [Help]
### Summary

First, thanks for your time and your help. I am using vercel.json to set a Content Security Policy (CSP) for a NextJS build. However, when I update that CSP, no matter how I set the level of caching across the site `/(.*)` or `/service-worker.js`: `no-cache`, the CSP remains the old one, unless the user on the client-side empties their browser's cache for the domain. This means the CSP never updates unless action is taken, there seems no expiry before it updates, which is not a desired nor acceptable behaviour.

Despite adding sha's and nonces to the CSP in the vercel.json, still no change.

The desired behaviour would be that the CSP updates when altered.
Please let me know how to do this, or what I am missing.

vercel.json looks like this:

```{
  "headers": [
    {
      "source": "/(.*)",
      "headers": [
        {
          "key": "Content-Security-Policy",
"value": "… removed for brevity script-src 'self' 'nonce-AbCdeFgH12iJkLmN' 'sha256-newHash';"
},
        {
          "key": "X-Content-Type-Options",
          "value": "nosniff"
        },
        {
          "key": "X-Frame-Options",
          "value": "DENY"
        },
        {
          "key": "X-XSS-Protection",
          "value": "1; mode=block"
        },
        { "key": "Access-Control-Allow-Credentials", "value": "true" },
        { "key": "Access-Control-Allow-Origin", "value": "*" },
        {
          "key": "Access-Control-Allow-Methods",
          "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT"
        },
        {
          "key": "Access-Control-Allow-Headers",
          "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
        }
      ]
    },
    {
      "source": "/service-worker.js",
      "headers": [
        {
          "key": "Content-Security-Policy",
          "value": "connect-src *;"
        },
        {
          "key": "Cache-Control",
          "value": "public, max-age=3600, stale-while-revalidate=604800"
        }
      ]
    },
],
  "crons": [
    {
      "path": "/api/cron",
      "schedule": "0 0 * * *"
    }
  ]
}```

### Example

_No response_

### Steps to Reproduce

1. Get any NextJS build that calls a 3rd-party site e.g. YouTube (youtube.com).
2. Implement a strict CSP, set in vercel.json:
```
  "headers": [
    {
      "source": "/(.*)",
      "headers": [
        {
          "key": "Content-Security-Policy",
"value": "default-src 'self'; connect-src 'self' blob:  img-src 'self' 'self' blob: data: https://*; media-src 'self' blob: https://*; object-src 'none'; style-src 'self' 'unsafe-inline'; base-uri 'none'; frame-ancestors 'self'; form-action 'self'; script-src-elem 'self' https://* https://vercel.live/* ; frame-src 'self' https://* https://vercel.live/ https://vercel.com;"
},
```
3. Deploy NextJS build to vercel. 
4. Visit deployed site's URL in browser.
5. Update CSP in vercel.json to include 3rd-party site's domain e.g. youtube.com and push changes to vercel.
6. Re-visit domain from same browser. Despite updating CSP, old version is used.
7. Add a sha and nonce to the end of the CSP's key in vercel.json to test if updates, but should act as before; cached on browser.

**Comment by ashbeech** [comment]
Have I not described the issue objectively or accurately enough? Let me know if more info's required to answer. Cheers.

**Comment by salazarm** [comment]
I also noticed this issue
